smartsim/_core/mli/infrastructure/storage/featurestore.py

```diff
@@ -28,6 +28,16 @@
 from abc import ABC, abstractmethod
 
 
+class FeatureStoreKey:
+    """A key-value pair enabling retrieval of an item in a feature store"""
+
+    def __init__(self, key: str, descriptor: str) -> None:
+        self.key = key
```

Comment:
Are we no longer using the ``:param`` syntax?

Reply:
brain fog on my part. i swapped to using `pydantic.BaseModel` so i could define min-length fields, skip writing a constructor, and have the nice property docstrings.

great catch